File: ShiftSync.Web/src/types/shiftsync/userShift.ts
import { User } from './user';
import { Shift } from './shift';

export interface UserShift {
  id: number;
  userId: string;
  shiftId: number;
  date: string; // "YYYY-MM-DD"
  user?: User;
  shift?: Shift;
}

export interface CreateUserShiftDto {
  userId: string;
  shiftId: number;
  date: string;
}

export interface UserShiftFilter {
  pageNumber?: number;
  pageSize?: number;
  userId?: string;
  shiftId?: number;
  dateFrom?: string;
  dateTo?: string;
}

File: ShiftSync.API/Models/Entities/UserShift.cs
namespace ShiftSync.API.Models.Entities;

public class UserShift
{
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;
    public ApplicationUser User { get; set; } = null!;

    public int ShiftId { get; set; }
    public Shift Shift { get; set; } = null!;

    public DateTime Date { get; set; }


    public ICollection<Attendance> Attendances { get; set; } = new List<Attendance>();
}
